jwst/associations/lib/rules_level3.py

```diff
@@ -273,7 +290,17 @@ def __init__(self, *args, **kwargs):
         self.constraints = Constraint([
             Constraint_Target(),
             Constraint_IFU(),
-        ])
+            Constraint(
+                [
+                    Constraint_TSO(),
+                    DMSAttrConstraint(
+                        name='patttype',
+                        sources=['patttype'],
+                        value=['2_point|4_point_nod|along_slit_nod'],
```

Comment:
The versions of these strings with underscores is what appears in PPS DB tables, but in our FITS keywords those underscores will be hyphens/dashes.

Reply:
So, question, when do they get converted? This is comparing to what appears in the pools. Will the pools get populated with the FITS-version or the PPS-version?

Reply:
I believe the pools are populated from the FITS keyword values, but we should confirm with SDP (i.e. Mike Swam).

Reply:
Mike S. confirms that with the delivery of b7.3 the FITS keyword and the asn pool entries will use dashes (-) and not underscores (_) in the PATTTYPE keyword. Should this be a new ticket?

Reply:
If any new tests in this PR do not require PATTTYPE, then it is probably best as a separate issue, since it will affect a number of other tests. However, if any new tests are implemented in this PR that require PATTTYPE, it is probably best to handle in this PR because might as well get it right in the first place.

Reply:
@ddavis-stsci Was a ticket filed?

Reply:
I just filed the ticket JP-664